### PR TITLE
Clean default parameters, refactor CNS scripts and tweak examples

### PR DESCRIPTION
**Pull request by joaomcteixeira**
@amjjbonvin here's the branch to clean the `defaults.cfg` file plus attempting to implement what we discussed about types. It goes (should)  as follow:

- numbers should be numbers, not strings
- parameters that are numbers that you DONT want to write in the CNS should be `param = nan`, without quotes.
- parameters that are strings that you don't want to write to the CNS should be empty strings `param = ""`
- Remove every `None` you find around
- For now, paths that are not to be written to CNS should be empty strings as well.

Give it a try, let me know the feeling.

### CODE REVIEW

File: examples/docking-protein-protein/docking-protein-protein.cfg
Comment (joaomcteixeira) [inline examples/docking-protein-protein/docking-protein-protein.cfg:7]: @amjjbonvin  Care adding these paths. Use `cns_exec = ""` will read the `bin` installation by default.

File: examples/docking-protein-protein/docking-protein-protein.cfg
Comment (joaomcteixeira) [inline examples/docking-protein-protein/docking-protein-protein.cfg:34]: Now is the perfect time to enforce `./` as in `./data` for paths if you wish. Is an idea, not a necessity,

**Comment by amjjbonvin** [comment]
Sorry - should comment it out again - was testing before if this was solving the CNS path problem

Remember - work in progress… :-)

**Comment by joaomcteixeira** [comment]
Can't understand why that error. `cns_exec` is reaching subprocess as an empty string. I can't reprocue that here. Can you add a `print(cns_exec)` [here in this file](https://github.com/haddocking/haddock3/blob/1555fd00e3fd4f70ccbb406b7b0ed9d6aeeca1f1/src/haddock/core/defaults.py#L16) and run `python defaults.py` and tell me what it prints?

**Comment by joaomcteixeira** [comment]
Hi @amjjbonvin 
I made a prototype code to identify the parameter blocks from the `deafults.cfg`. The code might not be complete or a bit buggy in some edge cases but may allow you to continue cleaning the `defaults.cfg` until I am back to the office (Monday). All cases I could think about for now are working.

An expandable parameter block is a group of parameters that share `first` and `_1`, for example the `c3sym` block. Also, the keys for each block are identified, for example `sta1`, `end1` etc, meaning if a user writes `c3sym_lala_1` results in a controlled error.

The code thus confirms the new block (for example `c3sym_*_2`) has the expected number of parameters.

There are some parameters that might break this rule, for example the `start_rair_*`. If these are blocks, could you rename those to `rair_1_start_1`, `rair_1_end_1` for example so that the block is defined by `rair_1` and not by `start`? Otherwise, how should we read these blocks?

Try it out, and let me know your feelings.

**Comment by amjjbonvin** [comment]
Should be doable and makes sense. Will take into account when further cleaning the default.cfg files

> There are some parameters that might break this rule, for example the start_rair_*. If these are blocks, could you rename those to rair_1_start_1, rair_1_end_1 for example so that the block is defined by rair_1 and not by start? Otherwise, how should we read these blocks?
>

**Comment by amjjbonvin** [comment]
Check the slack discussion - the expandable parameter implementation / check should allow for more flexibility following simply the following rule:

<unique key><anything>_1

And anything could even be empty.

File: examples/docking-protein-DNA/docking-protein-DNA-mdref.cfg
Comment (joaomcteixeira) [inline examples/docking-protein-DNA/docking-protein-DNA-mdref.cfg:25]: FYI, haddock3 config reader accepts both single and double quotes

**Review by amjjbonvin — APPROVED** [review]
I have been working a lot on the CNS parts and the defaults.cfg files. Everything working fine now. Time to merge it!

But @rvhonorato should review it as well.

File: src/haddock/gear/config_reader.py
Comment (rvhonorato) [inline src/haddock/gear/config_reader.py:52]: maybe also good to capture `NaN`?

File: src/haddock/modules/refinement/emref/cns/emref.cns
Comment (rvhonorato) [inline src/haddock/modules/refinement/emref/cns/emref.cns:414]: was this a hidden feature? 😉

File: src/haddock/modules/refinement/emref/cns/symmultimer.cns
Comment (rvhonorato) [inline src/haddock/modules/refinement/emref/cns/symmultimer.cns:28]: what was the purpose of this `decode`?

File: src/haddock/modules/refinement/flexref/cns/flexref.cns
Comment (rvhonorato) [inline src/haddock/modules/refinement/flexref/cns/flexref.cns:30]: The name of the parameters are more intuitive like this, we need to remember when writing the documentation that these are different from 2.x

File: src/haddock/modules/refinement/flexref/defaults.cfg
Comment (rvhonorato) [inline src/haddock/modules/refinement/flexref/defaults.cfg:184]: Does this need to be NONE for CNS? else would be less prone to confusion if it was `""`

File: src/haddock/modules/sampling/rigidbody/cns/rigidbody.cns
Comment (rvhonorato) [inline src/haddock/modules/sampling/rigidbody/cns/rigidbody.cns:26]: This has no use anymore, right?

File: src/haddock/modules/scoring/emscoring/cns/calc_free-ene.cns
Comment (rvhonorato) [inline src/haddock/modules/scoring/emscoring/cns/calc_free-ene.cns:16]: Just to confirm if this was not removed by mistake

File: src/haddock/modules/scoring/emscoring/cns/scale_inter_final.cns
Comment (rvhonorato) [inline src/haddock/modules/scoring/emscoring/cns/scale_inter_final.cns:23]: won't this impact the calculation?

File: src/haddock/modules/scoring/emscoring/cns/scoring.cns
Comment (rvhonorato) [inline src/haddock/modules/scoring/emscoring/cns/scoring.cns:148]: Isn't this needed to finish the calculation?

File: tests/test_libcns.py
Comment (rvhonorato) [inline tests/test_libcns.py:36]: *typo here

**Review by rvhonorato — COMMENTED** [review]
A big one! Have some small comments, just to confirm a few things.

File: examples/docking-protein-DNA/docking-protein-DNA-mdref.cfg
Comment (amjjbonvin) [reply examples/docking-protein-DNA/docking-protein-DNA-mdref.cfg:25]: For consistency I switched everywhere to double quotes

File: src/haddock/gear/config_reader.py
Comment (amjjbonvin) [reply src/haddock/gear/config_reader.py:52]: Why not indeed

File: src/haddock/modules/refinement/emref/cns/emref.cns
Comment (amjjbonvin) [reply src/haddock/modules/refinement/emref/cns/emref.cns:414]: Yes it was...

File: src/haddock/modules/refinement/emref/cns/symmultimer.cns
Comment (amjjbonvin) [reply src/haddock/modules/refinement/emref/cns/symmultimer.cns:28]: Before those variables were define with double quotes making them strings. CNS has a decode/encode function to convert strings to numbers and vice-versa

File: src/haddock/modules/refinement/flexref/cns/flexref.cns
Comment (amjjbonvin) [reply src/haddock/modules/refinement/flexref/cns/flexref.cns:30]: Will have to document all variables in the yaml file once we convert the cfg to yaml

File: src/haddock/modules/refinement/flexref/defaults.cfg
Comment (amjjbonvin) [reply src/haddock/modules/refinement/flexref/defaults.cfg:184]: none does has a meaning... Wouldn't change the behaviour since we only actually check in the code if that variable is equal to "alpha" "alphabeta" or "all"

File: src/haddock/modules/sampling/rigidbody/cns/rigidbody.cns
Comment (amjjbonvin) [reply src/haddock/modules/sampling/rigidbody/cns/rigidbody.cns:26]: `ridigmini` has indeed no use anymore. You simply don't call that module if you want to skip it.  One example of this is the `refine-complex` example.

But what if we want to dock during the flexible refinement stage (we did it in the past for some special cases)?  We might need in that case a `randorient` module after `topoaa`

File: src/haddock/modules/scoring/emscoring/cns/scale_inter_final.cns
Comment (amjjbonvin) [reply src/haddock/modules/scoring/emscoring/cns/scale_inter_final.cns:23]: Nope - we only calculate the intermolecular interactions and the weight for scoring should simply be 1. I removed the interaction matrix from the `default.cfg` file to simplify things.

File: src/haddock/modules/scoring/emscoring/cns/scoring.cns
Comment (amjjbonvin) [reply src/haddock/modules/scoring/emscoring/cns/scoring.cns:148]: No - already included in `calc_free-ene.cns` - was redundant.

File: src/haddock/modules/sampling/rigidbody/cns/rigidbody.cns
Comment (rvhonorato) [reply src/haddock/modules/sampling/rigidbody/cns/rigidbody.cns:26]: allright! seems quite specific, but now more then ever we have the "flexibility" to do it! 😄

**Review by rvhonorato — APPROVED** [review]

**Comment by joaomcteixeira** [comment]
@amjjbonvin what do to with the `parameter_1` case? Should this be a group also? I imagine not because you accepted the PR. But just want to make sure.

I am going to write the documentation and some tests, and after that I will merge.

File: src/haddock/gear/config_reader.py
Comment (joaomcteixeira) [reply src/haddock/gear/config_reader.py]: Done

**Comment by amjjbonvin** [comment]
Not needed indeed (for now)

**Pull request merged**
